src/base/record.rs

```diff
@@ -1479,6 +1496,51 @@ impl Ttl {
             .map(Ttl::from_secs)
             .map_err(Into::into)
     }
+
+    pub fn pretty(&self) -> impl fmt::Display {
+        struct Inner {
+            inner: Ttl,
+        }
+
+        impl fmt::Display for Inner {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                let days = self.inner.as_days();
+                let weeks = days / 7;
+                let days = days % 7;
+                let hours = self.inner.as_hours() % 24;
+                let minutes = self.inner.as_minutes() % 60;
+                let seconds = self.inner.as_secs() % 60;
+
+                let mut first = true;
+                for (n, unit) in [
+                    (weeks, "weeks"),
+                    (days, "days"),
+                    (hours as u16, "hours"),
+                    (minutes as u16, "minutes"),
+                    (seconds as u16, "seconds"),
+                ] {
+                    if n == 0 {
+                        continue;
+                    }
+                    if first {
+                        write!(f, " ")?;
+                    }
+                    write!(f, "{n}{unit}")?;
```

Comment:
To be clear, the format here is `3weeks 2hours 1minutes`?  That doesn't seem especially nice, unless it's for use in a particular format.  Either way, some documentation should be provided to show what the result will look like.

Reply:
I'd argue that we should have a separate method if we want compatibility with something specific.  You could name this `dig_display()` or something if that's exactly what you want to use it for.

Reply:
Remember that this is for a comment in the zonefile format (hence also `pub(crate)`)